frame/curve-amm/src/lib.rs

```diff
@@ -658,20 +604,19 @@ pub mod pallet {
 		}
 		fn exchange(
 			who: &Self::AccountId,
-			pool_id: PoolId,
-			i_token: PoolTokenIndex,
-			j_token: PoolTokenIndex,
+			pool_id: T::PoolId,
+			i_token: T::PoolTokenIndex,
+			j_token: T::PoolTokenIndex,
 			dx: Self::Balance,
 			min_dy: Self::Balance,
 		) -> Result<(), DispatchError> {
 			let prec = T::Precision::get();
```

Comment:
what will be weight of exchange? if it iterats over vec, so with time it can became very big

Reply:
how curve on substrate solves that?

Reply:
https://github.com/equilibrium-eosdt/equilibrium-curve-amm/tree/master/pallets/equilibrium-curve-amm/src this may have some answers on that.

Reply:
but we are anyway going to get rid of Vec.